service/common/src/main/java/org/apache/polaris/service/catalog/io/FileIOFactory.java

```diff
@@ -51,7 +51,7 @@ public interface FileIOFactory {
    * @return a configured FileIO instance.
    */
   FileIO loadFileIO(
-      @Nonnull RealmContext realmContext,
```

Comment:
For cleanup tasks, do we have CallContext since they are running in background threads.

Reply:
Sry, just read through the code, we now construct a `CallContext` in `TableCleanupTaskHandler`.

Reply:
Right, I think in general we'll need to better formalize a concept of a "SystemCallContext", and 99% of the code should still be able to operate on a CallContext passed in through method parameters, etc., without caring where the CallContext came from.

The initial construction would either come from request handlers and their injected dependencies, or explicit construction from some "system-privileged" type of thing.